pallets/aleph/src/lib.rs

```diff
@@ -123,6 +134,23 @@ pub mod pallet {
     pub(super) type FinalityScheduledVersionChange<T: Config> =
         StorageValue<_, VersionChange, OptionQuery>;
 
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_finalize(block_number: T::BlockNumber) {
+            if let Some(session_change_block) =
+                T::NextSessionRotation::estimate_next_session_rotation(block_number).0
+            {
+                if session_change_block == block_number + 1u32.into() {
```

Comment:
@DamianStraszak wont there be a problem here because of +-1 between abft and block production committee session block start?

Reply:
I think it's OK, as long as this thing `T::NextSessionRotation::estimate_next_session_rotation(block_number).0` is indeed always returning `session_period*k`.

The alephBFT change would then be included in block `900*k + 899` and the Aura change in `900*k+900`, which is fine. We cannot include the alephBFT change in block `900k+900` because that's actually one block to late, and the new committee would already sign it.